.pipelines/arc-register-extension-all-regions-stable.sh

```diff
@@ -0,0 +1,108 @@
+RELEASE_TRAINS_PREVIEW_PATH=${RELEASE_TRAINS_PREVIEW_PATH}
+RELEASE_TRAINS_STABLE_PATH=${RELEASE_TRAINS_STABLE_PATH}
+IS_CUSTOMER_HIDDEN=$IS_CUSTOMER_HIDDEN
+
+PACKAGE_CONFIG_NAME="${PACKAGE_CONFIG_NAME:-microsoft.azuremonitor.containers-pkg022022}"
+API_VERSION="${API_VERSION:-2021-05-01}"
+METHOD="${METHOD:-put}"
+REGISTRY_PATH_CANARY_PREVIEW="https://mcr.microsoft.com/azuremonitor/containerinsights/canary/preview/azuremonitor-containers"
+REGISTRY_PATH_CANARY_STABLE="https://mcr.microsoft.com/azuremonitor/containerinsights/canary/stable/azuremonitor-containers"
+REGISTRY_PATH_PROD_STABLE="https://mcr.microsoft.com/azuremonitor/containerinsights/prod1/stable/azuremonitor-containers"
+
+REGISTER_REGIONS_BATCH=($REGISTER_REGIONS_BATCH)
+
+echo "Start arc extension registration, REGISTER_REGIONS is $REGISTER_REGIONS_CANARY, RELEASE_TRAINS are $RELEASE_TRAINS_PREVIEW_PATH; $RELEASE_TRAINS_STABLE_PATH, PACKAGE_CONFIG_NAME is $PACKAGE_CONFIG_NAME, API_VERSION is $API_VERSION, METHOD is $METHOD"
+
+# Create JSON request body
+cat <<EOF > "request.json"
+{
+    "artifactEndpoints": [
+        {
+            "Regions": [
+                $REGISTER_REGIONS_CANARY
+            ],
+            "Releasetrains": [
+                $RELEASE_TRAINS_PREVIEW_PATH
+            ],
+            "FullPathToHelmChart": "$REGISTRY_PATH_CANARY_PREVIEW",
+            "ExtensionUpdateFrequencyInMinutes": 60,
+            "IsCustomerHidden": $IS_CUSTOMER_HIDDEN,
+            "ReadyforRollout": true,
+            "RollbackVersion": null,
+            "PackageConfigName": "$PACKAGE_CONFIG_NAME"
+        },
+EOF
+
+cat <<EOF >> "request.json"
+        {
+            "Regions": [
+                $REGISTER_REGIONS_CANARY
+            ],
+            "Releasetrains": [
+                $RELEASE_TRAINS_STABLE_PATH
+            ],
+            "FullPathToHelmChart": "$REGISTRY_PATH_CANARY_STABLE",
+            "ExtensionUpdateFrequencyInMinutes": 60,
+            "IsCustomerHidden": $IS_CUSTOMER_HIDDEN,
+            "ReadyforRollout": true,
+            "RollbackVersion": null,
+            "PackageConfigName": "$PACKAGE_CONFIG_NAME"
+        },
+EOF
+
+cat <<EOF >> "request.json"
+        {
+            "Regions": [
+                $REGISTER_REGIONS_BATCH
+            ],
+            "Releasetrains": [
+                $RELEASE_TRAINS_STABLE_PATH
+            ],
+            "FullPathToHelmChart": "$REGISTRY_PATH_PROD_STABLE",
+            "ExtensionUpdateFrequencyInMinutes": 60,
+            "IsCustomerHidden": $IS_CUSTOMER_HIDDEN,
+            "ReadyforRollout": true,
+            "RollbackVersion": null,
+            "PackageConfigName": "$PACKAGE_CONFIG_NAME"
+        },
+EOF
+
+sed -i '$ s/.$//' request.json
+
+cat <<EOF >> "request.json"
+    ]
+}
+EOF
+
+cat request.json | jq
+
+# Send Request
+SUBSCRIPTION=${ADMIN_SUBSCRIPTION_ID}
+RESOURCE_ID=${RESOURCE_ID}
+az login --service-principal --username=${SPN_CLIENT_ID} --password=${SPN_SECRET} --tenant=${SPN_TENANT_ID}
```

Comment:
what's the reason we are not using managed identity here?

Reply:
chatted with arc team, is now registering new tenant and client ids in extension config for trying msi

Reply:
after discussion   will migrate to using ev2 and then the existing msi